crypto/openstark/src/verifier.rs

```diff
@@ -21,7 +21,7 @@ pub fn check_proof<Public>(
     params: &ProofParams,
     trace_cols: usize,
     trace_len: usize,
-) -> bool
+) -> Result<(), &'static str>
```

Comment:
I think this should just return `Result<(), MerkleTreeError>`? You can then put the strings directly in the error, or use an enum, and derive Debug?

Reply:
Verifier can have other errors than the MerkleTree ones, so we need nested errors here. Rust has standard ways of handling this that we should use.